### DISCUSSION TITLE
Pattern matching without return value

### DISCUSSION THREAD

**Discussion by WashingtonARamos** [dotnet/csharplang#5896] [Language Ideas] [answered]
Currently in C# we can pattern match against variables like so:

```C#
var myStr = "a";
var newStr = myStr switch 
{
    "a" => "a+",
    "b" => "b+",
    _ => "Nothing"
};
```

My problem here is the need to return something. Sometimes I just want to evaluate some variable and do something based on it's value, so while I can use a discard (`_`) to simply not use the return variable at all I can't do something like this:

```C#
// SaySomething() and SaySomethingElse() are void
var val = 5;
val switch
{
    < 7 => SaySomething(),
    >= 7 => SaySomethingElse()
};
```

Is there a reason why we can't have "non-returning" pattern matching with switch? I think the syntax is simpler than doing:

```C#
var val = 5;
switch(val)
{
    case < 7:
        SaySomething();
        break;
    case >= 7:
        SaySomethingElse();
        break;
}
```

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/issues/3038
